src/openforms/forms/migrations/0079_replace_cosign_component.py

```diff
@@ -8,25 +8,22 @@
 def replace_old_cosign_component(apps, schema_editor):
     FormDefinition = apps.get_model("forms", "FormDefinition")
 
-    form_definitions = FormDefinition.objects.all()
+    # Selects all definitions containing a component of type 'coSign' (including nested)
+    form_definitions = FormDefinition.objects.raw(
+        """
+    SELECT * FROM forms_formdefinition
+    WHERE configuration  @? '$.** ? (@.type == "coSign")'
+    """
+    )
 
-    form_definitions_to_update = []
     for form_definition in form_definitions:
-        updated_form_definition = False
         for comp in iter_components(configuration=form_definition.configuration):
             if comp["type"] != "coSign":
                 continue
 
             comp.update({"type": "cosign", "validateOn": "blur"})
```

Comment:
@pi-sigma Like these nested for -loops: the looping can be named explicitly when written in a functional style:
```python
configurations = (d.configuration for d in form_definitions)
components = (iter_components(conf) for conf in configurations) 
cosign_components = (comp for comp in components if comp["type"] == "coSign")
# but then it breaks down, because we need to do mutation, which doesn't really fit the paradigm that well
for c in cosign_components:
    c.update(...)

# alternatively this  
components = map(iter_components, configurations)
cosign_components = filter(lambda c: c["type"] == "coSign", components)
# you could trigger the execution with a reduction where you ignore the accumulator:
functools.reduce(lambda acc, comp: comp.update(...), cosign_components)  # blegh
```
Arguably this is even more "step by step" than imperative code. The difference is that here, execution only happens in the last step and that you explicitly deal with the whole set of inputs instead of one at a time.